borsh-derive-internal/src/attribute_helpers.rs

```diff
@@ -42,6 +46,81 @@ pub(crate) fn contains_skip(attrs: &[Attribute]) -> bool {
     attrs.iter().any(|attr| attr.path() == SKIP)
 }
 
+pub fn check_item_attributes(derive_input: &DeriveInput) -> Result<(), TokenStream> {
+    for attr in &derive_input.attrs {
```

Comment:
```suggestion
    for attr in &derive_input.attrs {
        if attr.path().is_ident("borsh_skip") {
            return Err(TokenStream::from(
                syn::Error::new(
                    derive_input.ident.span(),
                    "`borsh_skip` is not allowed as derive input attribute",
                )
                .to_compile_error(),
            ));
        }
```
see paired comment for test below

Reply:
you've actually silenced test failure in this [commit](https://github.com/near/borsh-rs/pull/183/commits/3a9a3144003e9dea033dc12dc471bdaa9c7f5295), as you did before with `#[should_panic]`, not fixed it